constraints.txt

```diff
@@ -1,4 +1,4 @@
 numpy>=1.20,<2.0
-ipython<8.13;python_version<'3.9'
+ipython<8.13;python_version<'3.10'
```

Comment:
Is this change due to bumping the version we use from 3.8 to 3.9 and we just did not have the constraint right before in that it really applied to more than just 3.8

Reply:
My understanding of `<'3.9'` is that `ipython` will pick `3.8`, which is the only allowed version strictly lower than `3.9`. So, to allow `3.9` I raised `<'3.9'` to `<'3.10'`. Is this correct?

Reply:
This is a constraint (limits) over what is installed when it's needed to be installed. So that basically says if ipython is installed it should be less than version < 8.13 but with the caveat that this only for when the python version its being installed into is less than that value. So before it would have installed some version of ipython < 8.13 when using 3.8 and was not constrained in other cases so it would have used whatever version it got resolved to, which is likely much newer. Now with bumping the Python version there it will be constraining it to < 8.13 in Python 3.9, which it was not before. But then I do not know which jobs exactly end up needing/using it - I imagine its for the notebook ones - which we would have run on 3.8 (and 3.12) before but not 3.9. You could try and see without that constraint.

Reply:
Ok, I'll try out the tests without `ipython<8.13;python_version<'3.10'` in the next commit.

Reply:
Just pushed this update to remove the ipython constraint.